pkg/plugin/processor/service.go

```diff
@@ -70,6 +75,16 @@ func (s *PluginService) NewProcessor(ctx context.Context, pluginName string, id
 	}
 }
 
+// RegisterStandalonePlugin registers a standalone processor plugin from the
+// specified path and returns the full name of the plugin or an error.
+func (s *PluginService) RegisterStandalonePlugin(ctx context.Context, path string) (string, error) {
+	fullName, err := s.standaloneReg.Register(ctx, path)
+	if err != nil {
+		return "", cerrors.Errorf("failed to register standalone processor: %w", err)
```

Comment:
We can return it, but know that it might still be empty, there's multiple steps where registering a standalone plugin can fail before we get the full name from the plugin. Also, the plugin name will already be included in a verbose error message, if we have managed to retrieve it (e.g. [here](https://github.com/ConduitIO/conduit/blob/3f1b24fa1fe2079aae51eea7415dad9262f6a4a1/pkg/plugin/processor/standalone/registry.go#L285)).

Reply:
Returning failure is ok because the case where that is useful is exactly the the register the already registered. The annoyance from the error is caused by the need to parse a string to grab the name (the API's consumer could break in case of changes in the way the error is returned).

Reply:
Done! If the function returns an error containing [`plugin.ErrPluginAlreadyRegistered`](https://github.com/ConduitIO/conduit/blob/371a58453e0f23f265db402f4f0c4110ea1da316/pkg/plugin/errors.go#L25) you can expect the full name to be populated with the name of the plugin (see [test](https://github.com/ConduitIO/conduit/blob/371a58453e0f23f265db402f4f0c4110ea1da316/pkg/plugin/processor/standalone/registry_test.go#L146)).

Reply:
Thank you so much!